hummingbot/client/hummingbot_application.py

```diff
@@ -1141,6 +1142,99 @@ def compare_balance_snapshots(self):
         lines = ["", "  Performance:"] + ["    " + line for line in str(df).split("\n")]
         self._notify("\n".join(lines))
 
+    def add_balances(self, market_symbol_pair, is_base, is_starting, base, quote):
+        """ Adds the balance of either the base or the quote in the given market symbol pair token to the corresponding
+        tuple pair.
+
+        :param market_symbol_pair: type market_symbol_pair
+        :param is_base: boolean indicating whether to add base amount of token or quote
+        :param is_starting: boolean indicating whether to calculating starting amount of token or current
+        :param base: tuple that is the base currency token pair
+        :param quote: tuple that is the quote currency token pair
+        """
+        market_name = market_symbol_pair.market.name
+
+        if is_base:
+            asset_tuple = base
+            asset = market_symbol_pair.base_asset
+        else:
+            asset_tuple = quote
+            asset = market_symbol_pair.quote_asset
+
+        if is_starting:
+            amount = self.starting_balances.get(asset).get(market_name)
+        else:
+            amount = self.balance_snapshot().get(asset).get(market_name)
+
+        asset_list = list(asset_tuple)
+        if len(asset_list) == 0:
+            asset_list.insert(0, asset)
+            asset_list.insert(1, float(amount))
+        else:
+            if asset_list[0] == asset:
+                asset_list[1] += float(amount)
+            else:
+                erc = ExchangeRateConversion.get_instance()
+                asset_list[1] += float(erc.convert_token_value(
+                    amount=float(amount),
+                    from_currency=asset,
+                    to_currency=asset_list[0]
+                ))
+
+        asset_tuple = tuple(asset_list)
+        if is_base:
+            base = asset_tuple
+        else:
+            quote = asset_tuple
+
+        return base, quote
+
+    def analyze_performance(self):
+        """ Determine the profitability of the trading bot. """
+        if len(self.starting_balances) == 0:
+            self._notify("  Performance analysis is not available before bot starts")
+            return
+
+        # Each of these is in the format (currency, amount)
+        starting_base = tuple()
+        starting_quote = tuple()
+        current_base = tuple()
+        current_quote = tuple()
+
+        for market_symbol_pair in self.market_symbol_pairs:
+            for is_base in [True, False]:
+                starting_base, starting_quote = self.add_balances(
+                    market_symbol_pair=market_symbol_pair,
```

Comment:
You don't need to, and shouldn't use keywords if the argument isn't a keyword argument. If the function's argument name is changed later, your call will break.